DDD-4.md

```diff
@@ -0,0 +1,348 @@
+# DDD-4: Offset Read/Write API
+
+## Motivation
+
+Debezium is a platform to stream changes executed on data by processing transaction logs of a database.
+As changes are processed, a position into the transaction log is maintained so that Debezium can safely resume on restarts.
+
+This position, often referred to as offsets, differs by connector.
+While the MySQL connector maintains a reference to the binlog filename and a position within the binlog, other connectors maintain state that is specific to that database vendor's implementation.
+Therefore, there is no single common structure for these offsets.
+
+Debezium UI is a web-based application that is meant to simplify the deployment and management of Debezium connectors.
+We want to improve the user experience with the UI by introducing the ability to perform the following tasks:
+
+* Start, Stop, Pause, and Resume Incremental Snapshots
+* Manipulate connector offsets
+
+Both of these tasks require the ability to read connector offsets and the latter requires the ability to write to the offsets.
```

Comment:
Why are they required by incremental snapshots? To get the state and see if it is running?

Reply:
Correct, we don't want to allow the user to stop an incremental snapshot if one isn't currently running.  While we do understand that sending such a signal is ignored (no-op), it's so we design a good UX.